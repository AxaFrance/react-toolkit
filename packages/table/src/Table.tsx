import React from 'react';
import Tr from './Tr';
import Td from './Td';
import Th from './Th';
import Header from './Header';
import Body from './Body';
import TableCore, { TableCoreProps } from './TableCore';

class Table extends React.PureComponent<TableCoreProps> {
  public static readonly Header = Header;

  public static readonly Body = Body;

  public static readonly Th = Th;

  public static readonly Tr = Tr;

  public static readonly Td = Td;

  public render() {
    return <TableCore {...this.props} />;
  }
}
export default Table;
