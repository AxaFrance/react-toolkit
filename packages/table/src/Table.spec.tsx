import React from 'react';
import { create } from 'react-test-renderer';
import Table from './Table';

describe('<DateInput>', () => {
  it('renders DateInput correctly', () => {
    const tree = create(
      <Table>
        <Table.Header>
          <Table.Tr>
            <Table.Th colSpan={3}>
              <span className="af-table-th-content">Some text</span>
            </Table.Th>
          </Table.Tr>
        </Table.Header>
        <Table.Body>
          <Table.Tr>
            <Table.Td rowSpan={2}>
              <span className="af-table-body-content">Some text</span>
            </Table.Td>
            <Table.Td>
              <b>Some balised text</b>
            </Table.Td>
            <Table.Td>
              <span className="af-table-body-content">Some text</span>
            </Table.Td>
          </Table.Tr>
        </Table.Body>
      </Table>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
