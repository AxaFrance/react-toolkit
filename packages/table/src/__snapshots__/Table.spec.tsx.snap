// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateInput> renders DateInput correctly 1`] = `
<table
  className="af-table"
>
  <thead
    className="af-table-thead"
  >
    <tr
      className="af-table__tr"
    >
      <th
        className="af-table__th"
        colSpan={3}
        rowSpan={null}
      >
        <span
          className="af-table-th-content"
        >
          Some text
        </span>
      </th>
    </tr>
  </thead>
  <tbody
    className="af-table__body"
  >
    <tr
      className="af-table__tr"
    >
      <td
        className="af-table__cell"
        colSpan={null}
        rowSpan={2}
      >
        <span
          className="af-table-body-content"
        >
          Some text
        </span>
      </td>
      <td
        className="af-table__cell"
        colSpan={null}
        rowSpan={null}
      >
        <b>
          Some balised text
        </b>
      </td>
      <td
        className="af-table__cell"
        colSpan={null}
        rowSpan={null}
      >
        <span
          className="af-table-body-content"
        >
          Some text
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
